specification/monitor/resource-manager/Microsoft.Insights/preview/2017-05-01-preview/subscriptionDiagnosticsSettings_API.json

```diff
@@ -0,0 +1,341 @@
+						},
+						"description": "Parameters supplied to the operation."
+					},
+					{
+						"$ref": "#/parameters/NameParameter"
+					}
+				],
+				"responses": {
+					"default": {
+						"description": "Error response describing why the operation failed.",
+						"schema": {
+							"$ref": "#/definitions/ErrorResponse"
+						}
+					},
+					"200": {
+						"description": "Successful request to create a subscription diagnostic setting",
+						"schema": {
+							"$ref": "#/definitions/SubscriptionDiagnosticSettingsResource"
+						}
+					}
+				},
+				"x-ms-examples": {
+					"Creates or Updates the subscription diagnostic setting": {
+						"$ref": "./examples/createOrUpdateSubscriptionDiagnosticSetting.json"
+					}
+				}
+			},
+			"delete": {
+				"tags": [
+					"SubscriptionDiagnosticSettings"
+				],
+				"operationId": "SubscriptionDiagnosticSettings_Delete",
+				"description": "Deletes existing subscription diagnostic settings for the specified resource.",
+				"parameters": [
+					{
+						"$ref": "#/parameters/SubscriptionIdParameter"
+					},
+					{
+						"$ref": "#/parameters/ApiVersionParameter"
+					},
+					{
+						"$ref": "#/parameters/NameParameter"
+					}
+				],
+				"responses": {
+					"default": {
+						"description": "Error response describing why the operation failed.",
+						"schema": {
+							"$ref": "#/definitions/ErrorResponse"
+						}
+					},
+					"200": {
+						"description": "Successful request to remove a subscription diagnostic setting"
+					},
+					"204": {
+						"description": "Successful request to remove a subscription diagnostic setting"
+					}
+				},
+				"x-ms-examples": {
+					"Deletes the subscription diagnostic setting": {
+						"$ref": "./examples/deleteSubscriptionDiagnosticSetting.json"
+					}
+				}
+			}
+		},
+		"/subscriptions/{subscriptionId}/providers/microsoft.insights/diagnosticSettings": {
+			"get": {
+				"tags": [
+					"SubscriptionDiagnosticSettings"
+				],
+				"operationId": "SubscriptionDiagnosticSettings_List",
+				"description": "Gets the active subscription diagnostic settings list for the specified subscriptionId.",
+				"parameters": [
+					{
+						"$ref": "#/parameters/SubscriptionIdParameter"
+					},
+					{
+						"$ref": "#/parameters/ApiVersionParameter"
+					}
+				],
+				"responses": {
+					"default": {
+						"description": "Error response describing why the operation failed.",
+						"schema": {
+							"$ref": "#/definitions/ErrorResponse"
+						}
+					},
+					"200": {
+						"description": "Successful request to get more information about subscription diagnostic setting",
+						"schema": {
+							"$ref": "#/definitions/SubscriptionDiagnosticSettingsResourceCollection"
+						}
+					}
+				},
+				"x-ms-examples": {
+					"Gets the subscription diagnostic setting": {
+						"$ref": "./examples/listSubscriptionDiagnosticSettings.json"
+					}
+				}
+			}
+		}
+	},
+	"definitions": {
+		"SubscriptionProxyOnlyResource": {
```

Comment:
It's redundant to add common definitions here. It can be referenced from another swagger spec, or better yet - [the common definitions swagger.](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/types.json)